middle_end/flambda2/to_cmm/to_cmm_shared.mli

```diff
@@ -86,20 +86,34 @@ val bound_parameters :
 val invalid :
   To_cmm_result.t -> message:string -> Cmm.expression * To_cmm_result.t
 
-type update_kind =
-  | Word_val
-  | Word_int
-  | Single of { reg : Cmm.float_width }
-  | Double
-  | Thirtytwo_signed
-  | Onetwentyeight_unaligned
+module Update_kind : sig
+  type t
+
+  val pointers : t
+
+  val tagged_immediates : t
+
+  val naked_int32s : t
+
+  val naked_int64s : t
+
+  val naked_floats : t
+
+  val naked_float32s : t
+
+  val naked_float32_fields : t
+
+  val naked_vec128s : t
+
+  val naked_vec128_fields : t
+end
 
 (** Make an update to a statically-allocated block. *)
```

Comment:
This comment should be updated to mention that this function only works if all elements stored in the block are of the same width. Alternatively, the code should be modified to track separately the position within the block where the update has to take place.

Reply:
Added comments on the update kinds